neuro_san/session/direct_agent_session.py

```diff
@@ -187,17 +188,20 @@ def streaming_chat(self, request_dict: Dict[str, Any]) -> Generator[Dict[str, An
                                          keep_alive_result=empty,
                                          keep_alive_timeout_seconds=10.0,
                                          umbrella_timeout=self.umbrella_timeout)
-        for message in generator.synchronously_iterate(self.invocation_context.get_queue()):
-
-            response_dict: Dict[str, Any] = copy(template_response_dict)
-            if message_filter.allow(message):
-                # We expect the message to be a dictionary form of chat.ChatMessage
-                if message_processor is not None:
-                    message_type: ChatMessageType = message.get("type")
-                    # Can modify message
-                    message_processor.process_message(message, message_type)
-                response_dict["response"] = message
-                yield response_dict
+        try:
+            for message in generator.synchronously_iterate(self.invocation_context.get_queue()):
+                response_dict: Dict[str, Any] = copy(template_response_dict)
+                if message_filter.allow(message):
+                    # We expect the message to be a dictionary form of chat.ChatMessage
+                    if message_processor is not None:
+                        message_type: ChatMessageType = message.get("type")
+                        # Can modify message
+                        message_processor.process_message(message, message_type)
+                    response_dict["response"] = message
+                    yield response_dict
+        finally:
+            with contextlib.suppress(Exception):
+                asyncio.run(chat_session.delete_resources())
 
```

Comment:
Duplication of logic from async_direct_agent_session.py for the case of sync iterators.